lib/zendesk_apps_support/package.rb

```diff
@@ -225,7 +225,9 @@ def iframe_only?
     private
 
     def legacy_non_iframe_app?
-      @non_iframe ||= locations.values.flat_map(&:values).any? { |l| l == LEGACY_URI_STUB }
+      iframe_urls = locations.values.flat_map(&:values)
+      return true if iframe_urls.empty?
+      @non_iframe ||= iframe_urls.any? { |l| l == LEGACY_URI_STUB }
```

Comment:
If the empty case is `true`, would it make it clearer to invert the `any?` logic and remove the early `return`? e.g.:

```
is_iframe_location = -> (loc) { loc != LEGACY_URI_STUB }
urls = locations.values.flat_map(&:values)
not urls.all?(&is_iframe_location)
```

Also the memoization here is a bit haphazard - it won't short-circuit when this method returns `false`, nor when we hit the early return. And in any case, the line to calculate `iframe_urls` is probably more expensive than the part that's being short-circuited. I'd vote to ditch the memoization unless it's worth the extra code to get it right.

Out of curiosity: can we have a mix of legacy and modern locations?